common/features/ens/helpers.ts

```diff
@@ -89,9 +89,10 @@ const modeMap: IModeMap = {
   [NameState.NotYetAvailable]: (_: IDomainData<NameState.NotYetAvailable>) => ({})
 };
 
-export function* resolveDomainRequest(name: string): SagaIterator {
-  const hash = ethUtil.sha3(name);
-  const nameHash = getNameHash(`${name}.eth`);
+export function* resolveDomainRequest(fullDomain: string): SagaIterator {
+  const domain = extractDomain(fullDomain);
+  const hash = ethUtil.sha3(domain);
+  const nameHash = getNameHash(fullDomain);
 
   const domainData: typeof ENS.auction.entries.outputType = yield call(makeEthCallAndDecode, {
```

Comment:
Perhaps this should only be done if the name is a direct subdomain of 'eth'?

Reply:
I thought it wouldn't be necessary to force the 'eth' tld since there are already checks [here](https://github.com/MyCryptoHQ/MyCrypto/blob/develop/common/components/AddressFieldFactory/AddressInputFactory.tsx#L89) and [here](https://github.com/MyCryptoHQ/MyCrypto/blob/develop/common/features/transaction/sagas.ts#L47) that make sure the name to resolve is under 'eth'.

Besides, current implementation forces an auction check regardless the tld is managed by an auction registrar or no. Future works (e.g. using a `tld => resolve function` map) could use the actual tld.